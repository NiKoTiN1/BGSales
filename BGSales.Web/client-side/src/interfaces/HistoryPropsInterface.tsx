export default interface HistoryPropsInterface {
  push(url: string): void;
  goBack: any;
}
