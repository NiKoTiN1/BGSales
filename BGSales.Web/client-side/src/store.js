import { createStore, combineReducers, applyMiddleware } from "redux";
import reducer from "./reducers";
import thunk from "redux-thunk";
var rootReducer = combineReducers({
    reducer: reducer,
});
var store = createStore(rootReducer, applyMiddleware(thunk));
export default store;
