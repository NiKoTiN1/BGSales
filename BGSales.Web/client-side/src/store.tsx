import { createStore, combineReducers, applyMiddleware } from "redux";
import reducer from "./reducers";
import thunk from "redux-thunk";

const rootReducer = combineReducers({
  reducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
