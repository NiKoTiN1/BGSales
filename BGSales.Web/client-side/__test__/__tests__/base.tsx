import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import AdvertiserPersonProfile from "../../src/components/advertiser-person-profile";
import AppHeader from "../../src/components/app-header";
import UserMenu from "../../src/components/user-menu";
import AppFooter from "../../src/components/app-footer";
import store from "../../src/store"
import React from 'react';
import HistoryPropsInterface from "../../src/interfaces/HistoryPropsInterface";

jest.mock("history");
jest.useFakeTimers()
describe("BGSales Base module", () => {
	const history: any = 0;

	it("check be truth base element footer ", () => {

		const wrapper = renderer.create(<Provider store={store}><AppFooter /> </Provider>);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});


	it("check be truth base element user menu ", () => {

		const wrapper = renderer.create(<Provider store={store}><UserMenu onClick={()=>{}} /> </Provider>);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});
});
