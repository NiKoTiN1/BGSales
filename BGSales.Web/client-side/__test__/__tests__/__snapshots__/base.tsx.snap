// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BGSales Base module check be truth base element footer  1`] = `
Array [
  <div
    className="footer"
  >
    <label
      className="footer__label"
    >
      The best project
    </label>
    <label
      className="footer__label"
    >
      rodinhatskevich@gmail.com
    </label>
    <label
      className="footer__label"
    >
      Â© Gatskevich Rodion, Biletsky Nikita
    </label>
    <div
      className="footer__background-images"
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
  </div>,
  " ",
]
`;

exports[`BGSales Base module check be truth base element user menu  1`] = `
Array [
  <div
    className="user-menu"
  >
    <div>
      <img
        alt=""
        className="user-menu__img"
        onClick={[Function]}
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
  </div>,
  " ",
]
`;
